src/js/common/components/Nav.jsx

```diff
@@ -1,18 +1,83 @@
 import React from 'react';
-import { IndexLink, Link } from 'react-router';
+import _ from 'lodash';
+import classnames from 'classnames';
 
+function lastPage(chapter) {
+  return chapter.pages.slice(-1)[0].path;
+}
+
+// Checks to see if a page's data dependencies are met
+// i.e., has the user filled out the required information?
+function isHidden(page, data) {
+  return page.depends !== undefined && _.matches(page.depends)(data) === false;
+}
+
+function determineChapterStyles(pageState, formChapter, currentUrl) {
+  return classnames(
+    { 'section-current': formChapter.pages.some(page => page.path === currentUrl) },
+    { 'section-complete': formChapter.pages.length > 0 && pageState[lastPage(formChapter)].complete }
+  );
+}
+
+function determinePageStyles(page, currentUrl, data) {
+  return classnames(
+    { 'sub-section-current': currentUrl === page.path },
+    { 'sub-section-hidden': isHidden(page, data) }
```

Comment:
I'm curious if you have any thoughts on hiding these pages vs filtering them out of the data entirely.